app/main/lib/shared_models/video_model.py

```diff
@@ -140,7 +137,7 @@ def search_by_context(self, context):
                 cmd = """
                   SELECT id, doc_id, url, folder, filepath, context, hash_value FROM videos
                 """
-            matches = db.session.execute(text(cmd), context_hash).fetchall()
+            matches = self.execute_command(text(cmd), context_hash)
```

Comment:
I'm not sure what you are requesting here. Are you asking to refactor to remove the lines above this into its own function?

Reply:
I was suggesting this as part of the refactor above: instead of the video model knowing how to speak sql, it could call a `get_videos_matching_hash` function in a database wrapper